packages/axios-extension/src/abap/catalog/v2-catalog-service.ts

```diff
@@ -138,8 +145,11 @@ export class V2CatalogService extends CatalogService {
      * @param filter.path filter by path
      * @returns service annotations
      */
-    protected async getServiceAnnotations({ id, title, path }: FilterOptions): Promise<ODataServiceV2Info[]> {
-        if (!this.entitySet) {
+    protected async getServiceAnnotations(
+        { id, title, path }: FilterOptions,
+        preloadCatalog = true
```

Comment:
Getting the correct entity set is not about preloading, it is about fining the correct services depending on the version of the ABAP backend.

Reply:
Looking at it again, I see your point. It is required for fetching the services, however, for fetching the annotation of a specific service you can always use `ServiceCollection`, so there is no need to add this preload parameter but you can remove the  `this.determineEntitySet` call from and always use `/ServiceCollection` as the base path.

Reply:
Done thanks